pymeasure/instruments/siglent/sdl1022x.py

```diff
@@ -0,0 +1,213 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2021 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_range, strict_discrete_set
+
+VALID_LOAD_MODES_DICT = {
+    "CURRent": "CURRent",
+    "CURRENT": "CURRent",
+    "CURR": "CURRent",
+    "VOLTage": "VOLTage",
+    "VOLTAGE": "VOLTage",
+    "VOLT": "VOLTage",
+    "POWer": "POWer",
+    "POWER": "POWer",
+    "POW": "POWer",
+    "RESistance": "RESistance",
+    "RESISTANCE": "RESistance",
+    "RES": "RESistance",
+}
+VALID_CURRENT_RANGES = [5, 30]
+VALID_VOLTAGE_RANGES = [36, 150]
+VALID_RESISTANCE_RANGES = ["LOW", "MIDDLE", "HIGH", "UPPER"]
+
+
+class Sdl1022X(Instrument):
+    """
+     Represents the Siglent SDL1000X Electronic Load.
+
+     .. code-block:: python
+
+        load = Sdl1022X("USB0::...")
+        load.reset()
+        load.load_mode = "RESISTANCE"
+        load.constant_resistance_load = 10  # Ohms
+        load.set_voltage_range("RESISTANCE", 5)  # Volts
+        load.input_state = "ON"
+        print(load.voltage)
+
+    """
+    voltage = Instrument.measurement(
+        "MEASure:VOLT:DC?",
+        "Voltage across the load in Volts"
+    )
+
+    current = Instrument.measurement(
+        "MEASure:CURR:DC?",
+        "Current across the load in Amps"
+    )
+
+    power = Instrument.measurement(
+        "MEASure:POW:DC?",
+        "Power across the load in Watts"
+    )
+
+    resistance = Instrument.measurement(
+        "MEASure:RES:DC?",
+        "Resistance across the load in Ohms"
+    )
+
+    external = Instrument.measurement(
+        "MEASure:EXT?",
+        "Voltage across the external inputs in Volts"
+    )
+
+    input_state = Instrument.control(
```

Comment:
Boolean properties are most clearly named if they correspond to a true/false adjective -- input_state_enabled? Yes! Set input state enabled to False, ...